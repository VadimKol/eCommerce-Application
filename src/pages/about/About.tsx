import classNames from 'classnames';

import { CommandBlock } from '@/components/command-block/CommandBlock.tsx';

import styles from './styles.module.scss';

const commandInfo = [
  {
    title: 'Ivan Zhirebny',
    classPhoto: 'imageIzy',
    link: 'https://github.com/izy-code',
    textInfo: {
      role: 'frontend-developer',
      contributions: 'CommerceTools, Tests, Routing, Detail page, Header, Breadcrumbs, React contexts, 404 page',
      'Short bio':
        ' Graduated from the Bauman Moscow State Technical University, completed the Frontend Developer program at the HTML Academy',
      'Soft stack': 'CSS, Sass, HTML, JavaScript, Java, TypeScript, Webpack, Vite, Jest, React',
    },
    github: 'izy-code',
    simpleText: [],
  },
  {
    title: 'Vadim Kolymbet',
    classPhoto: 'imageVadim',
    link: 'https://github.com/VadimKol',
    textInfo: {
      role: 'frontend-developer',
      contributions: 'CommerceTools, Tests, Login page, Catalog page, Cart page',
      'Short bio':
        'Graduated from Saratov State University, studied at the Faculty of Computer Science and Information Technologies and in the Department of Computer Security and Cryptography Theory.',
      'Soft stack': 'CSS, Sass, SQL, HTML, JavaScript, TypeScript, Webpack, Vite, Jest, React',
    },
    github: 'vadimkol',
    simpleText: [],
  },
  {
    title: 'Aleksandra Bodnar',
    classPhoto: 'imageAlex',
    link: 'https://github.com/BodnarAlex',
    textInfo: {
      role: ' frontend-developer, scrum-master',
      contributions: 'CommerceTools, Design, Registration page, Home page, About page, Profile page',
      'Short bio':
        'Graduated from the Immanuel Kant Baltic Federal University, studied at the Faculty of Business Informatics. Graduated from 3WC courses with a degree in Frontend-developer. Worked on PHP for 1C CRM.',
      'Soft stack': 'PHP, C++, Python, Wordpress, CSS, Sass, HTML, JavaScript, TypeScript, Webpack, Vite, React',
    },
    github: 'bodnaralex',
    simpleText: [],
  },
  {
    title: 'Rolling Scopes School',
    classPhoto: 'imageRss',
    link: 'https://rollingscopes.com/',
    textInfo: {},
    simpleText: [
      'RSSchool is an online educational community specializing in web development and programming. It offers comprehensive, well-structured courses designed for both beginners and experienced developers. Known for its high-quality education, RSSchool provides an extensive curriculum that covers everything from basic HTML, CSS, and JavaScript to advanced technologies like React and Node.js. The school is emphasis on project-based learning allows students to build real-world projects and create impressive portfolios.',
      'RSSchool thrives on a vibrant community of volunteers, including seasoned developers and alumni who mentor and support new learners.',
    ],
    github: '',
  },
];

export function About(): JSX.Element {
  return (
    <main className={classNames('main', styles.main)}>
      <div className={styles.container}>
        <h1 className="visually-hidden">About page</h1>
        <div className={styles.blocksInfo}>
          {commandInfo.map((info, index) => (
            <CommandBlock
              key={info.classPhoto}
              title={info.title}
              classPhoto={info.classPhoto}
              link={info.link}
              textInfo={info.textInfo}
              simpleText={info.simpleText}
              github={info.github}
              isEvenBlock={(index + 1) % 2 === 0}
            />
          ))}
        </div>
      </div>
    </main>
  );
}
