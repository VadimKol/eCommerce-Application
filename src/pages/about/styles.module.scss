.main {
  padding: 0 20px;
}

.container {
  display: flex;
  flex-direction: column;
  gap: 10px;
  width: 100%;
  min-width: 300px;
  max-width: 100%;
  margin: auto;
  padding: 28px 40px 30px;
  font-family: var(--primary-font);
  text-align: center;
  border-radius: 10px;
}


.title {
  margin-block: 0;
  text-shadow: var(--background-color) 2px 0 5px;
}

.blockInfo {
  display: flex;
  flex-flow: row nowrap;
  gap: 20px;
  justify-content: space-between;
  align-items: center;
  box-sizing: border-box;
  width: 100%;
  max-width: 1200px;
  margin: auto;
  padding: 20px 40px 10px;
  color: var(--primary-color);
  background:
    linear-gradient(var(--background-color) 0 0) padding-box,
    linear-gradient(45deg, var(--primary-color), var(--secondary-color)) border-box;
  border: 5px solid transparent;
  border-radius: 10px;
}

.blocksInfo {
  display: flex;
  flex-flow: column nowrap;
  gap: 10px;

  & .linkImage {
    color: var(--primary-color);
    background:
      linear-gradient(var(--background-color) 0 0) padding-box,
      linear-gradient(45deg, var(--primary-color), var(--secondary-color)) border-box;
    border: 5px solid transparent;
    border-radius: 50%;
  }
}

.blockInfo.rotate {
  flex-flow: row-reverse nowrap;
  background:
    linear-gradient(var(--background-color) 0 0) padding-box,
    linear-gradient(45deg, var(--secondary-color), var(--primary-color)) border-box;

  & .linkImage {
    background:
      linear-gradient(var(--background-color) 0 0) padding-box,
      linear-gradient(45deg, var(--secondary-color), var(--primary-color)) border-box;
  }
}


.imageIzy {
  background: url("../../assets/images/izy.jpg");
}

.imageVadim {
  background: url("../../assets/images/vadim.png");
}

.imageAlex {
  background: url("../../assets/images/alex.png");
}

.imageCommand.imageRss {
  background: url("../../assets/images/rss.png");
  background-position: center;
  background-size: 100%;
  background-repeat: no-repeat;
  border-radius: inherit;
}

.textInfo {
  display: flex;
  flex-flow: column nowrap;
  gap: 10px;
  justify-content: space-around;
}

.textInfo div {
  margin-bottom: 10px;
  line-height: 20px;
  text-align: left;
  text-indent: 20px;
}

.imageCommand {
  width: 300px;
  height: 300px;
  background-position: center;
  background-size: 100%;
  background-repeat: no-repeat;
  border-radius: 50%;
}

.blockRss {
  width: 100%;
  margin: 20px auto 0;
  text-align: center;
  background-color: var(--tertiary-color);
}


.link {
  text-decoration: none;
  color: var(--primary-color);
}

@media (width <= 800px) {
  .blockInfo {
    flex-flow: column nowrap;

    &.rotate {
      flex-flow: column nowrap;
    }
  }
}

@media (width <= 500px) {
  .container {
    margin: 0;
    padding: 0;
  }
}
