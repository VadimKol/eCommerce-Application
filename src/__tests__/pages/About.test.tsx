import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import { About } from '@/pages/about/About';

it('renders correctly', () => {
  const { container } = render(
    <BrowserRouter>
      <About />
    </BrowserRouter>,
  );
  expect(container).toMatchSnapshot();
});
