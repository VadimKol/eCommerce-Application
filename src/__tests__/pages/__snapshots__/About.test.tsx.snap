// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <main
    class="main main"
  >
    <div
      class="container"
    >
      <h1
        class="visually-hidden"
      >
        About page
      </h1>
      <div
        class="blocksInfo"
      >
        <div
          class="blockIntro"
        >
          <h2
            class="introTitle"
          >
            Team work
          </h2>
          <div
            class="introParagraph"
          >
            To collaborate effectively, we held regular meetings to share progress, look for solutions to problems, and find the best ideas.
          </div>
          <div
            class="introParagraph"
          >
            Aleksandra conceived and brought to life the overall look of the site, and the team subsequently refined the color palette and layout down to the smallest detail.
          </div>
          <div
            class="introParagraph"
          >
            Vadim was the first to work on queries to CommerceTools and at a scheduled meeting he taught the logic of how queries work, which subsequently allowed the entire team to write their queries without any problems.
          </div>
          <div
            class="introParagraph"
          >
            Ivan implemented a contextual system into our React project, which allowed the site to run without extra requests to the backend. He also shared his knowledge among the participants, allowing them to use it in the future without any problems.
          </div>
          <div
            class="introParagraph"
          >
            We implemented agile methods, including stand-ups every two days and sprint planning. When problems arose, we worked closely together to eliminate bottlenecks.
          </div>
          <div
            class="introParagraph"
          >
            To sum it up, our collaboration was like a symphony: diverse talents came together to create a remarkable store.
          </div>
        </div>
        <div
          class="blockInfo"
        >
          <a
            class="linkImage"
            href="https://github.com/izy-code"
          >
            <div
              class="imageCommand imageIzy"
            />
          </a>
          <div
            class="textInfo"
          >
            <a
              class="titleBlock"
              href="https://github.com/izy-code"
            >
              <h2>
                Ivan Zhirebny
                 
              </h2>
            </a>
            <div>
              <span
                class="infoTitle"
              >
                Github: 
              </span>
              <a
                class="link"
                href="https://github.com/izy-code"
              >
                izy-code
              </a>
            </div>
            <div>
              <span
                class="infoTitle"
              >
                Role: 
              </span>
              frontend-developer
            </div>
            <div>
              <span
                class="infoTitle"
              >
                Contributions: 
              </span>
              CommerceTools, Tests, Routing, Detailed product page, Error page, Header, Breadcrumbs, React contexts, 404 page
            </div>
            <div>
              <span
                class="infoTitle"
              >
                Short bio: 
              </span>
               Graduated from the Bauman Moscow State Technical University, completed the Frontend Developer program at the HTML Academy
            </div>
            <div>
              <span
                class="infoTitle"
              >
                Soft stack: 
              </span>
              <span
                class="softStackItem"
              >
                CSS
              </span>
              <span
                class="softStackItem"
              >
                Sass
              </span>
              <span
                class="softStackItem"
              >
                HTML
              </span>
              <span
                class="softStackItem"
              >
                JavaScript
              </span>
              <span
                class="softStackItem"
              >
                Java
              </span>
              <span
                class="softStackItem"
              >
                TypeScript
              </span>
              <span
                class="softStackItem"
              >
                Webpack
              </span>
              <span
                class="softStackItem"
              >
                Vite
              </span>
              <span
                class="softStackItem"
              >
                Jest
              </span>
              <span
                class="softStackItem"
              >
                React
              </span>
            </div>
          </div>
        </div>
        <div
          class="blockInfo rotate"
        >
          <a
            class="linkImage"
            href="https://github.com/VadimKol"
          >
            <div
              class="imageCommand imageVadim"
            />
          </a>
          <div
            class="textInfo"
          >
            <a
              class="titleBlock"
              href="https://github.com/VadimKol"
            >
              <h2>
                Vadim Kolymbet
                 
              </h2>
            </a>
            <div>
              <span
                class="infoTitle"
              >
                Github: 
              </span>
              <a
                class="link"
                href="https://github.com/VadimKol"
              >
                vadimkol
              </a>
            </div>
            <div>
              <span
                class="infoTitle"
              >
                Role: 
              </span>
              frontend-developer
            </div>
            <div>
              <span
                class="infoTitle"
              >
                Contributions: 
              </span>
              CommerceTools, Tests, Login page, Catalog page, Cart page, Repository setup
            </div>
            <div>
              <span
                class="infoTitle"
              >
                Short bio: 
              </span>
              Graduated from Saratov State University, studied at the Faculty of Computer Science and Information Technologies and in the Department of Computer Security and Cryptography Theory. Graduated layout courses from Frontendblok. Worked as a TSQL developer
            </div>
            <div>
              <span
                class="infoTitle"
              >
                Soft stack: 
              </span>
              <span
                class="softStackItem"
              >
                CSS
              </span>
              <span
                class="softStackItem"
              >
                Sass
              </span>
              <span
                class="softStackItem"
              >
                SQL
              </span>
              <span
                class="softStackItem"
              >
                HTML
              </span>
              <span
                class="softStackItem"
              >
                JavaScript
              </span>
              <span
                class="softStackItem"
              >
                TypeScript
              </span>
              <span
                class="softStackItem"
              >
                Webpack
              </span>
              <span
                class="softStackItem"
              >
                Vite
              </span>
              <span
                class="softStackItem"
              >
                Jest
              </span>
              <span
                class="softStackItem"
              >
                React
              </span>
            </div>
          </div>
        </div>
        <div
          class="blockInfo"
        >
          <a
            class="linkImage"
            href="https://github.com/BodnarAlex"
          >
            <div
              class="imageCommand imageAlex"
            />
          </a>
          <div
            class="textInfo"
          >
            <a
              class="titleBlock"
              href="https://github.com/BodnarAlex"
            >
              <h2>
                Aleksandra Bodnar
                 
              </h2>
            </a>
            <div>
              <span
                class="infoTitle"
              >
                Github: 
              </span>
              <a
                class="link"
                href="https://github.com/BodnarAlex"
              >
                bodnaralex
              </a>
            </div>
            <div>
              <span
                class="infoTitle"
              >
                Role: 
              </span>
               frontend-developer, scrum-master
            </div>
            <div>
              <span
                class="infoTitle"
              >
                Contributions: 
              </span>
              CommerceTools, Design, Registration page, Home page, About page, Profile page
            </div>
            <div>
              <span
                class="infoTitle"
              >
                Short bio: 
              </span>
              Graduated from the Immanuel Kant Baltic Federal University, studied at the Faculty of Business Informatics. Graduated from 3WC courses with a degree in Frontend-developer. Worked on PHP for 1C CRM
            </div>
            <div>
              <span
                class="infoTitle"
              >
                Soft stack: 
              </span>
              <span
                class="softStackItem"
              >
                PHP
              </span>
              <span
                class="softStackItem"
              >
                C++
              </span>
              <span
                class="softStackItem"
              >
                Python
              </span>
              <span
                class="softStackItem"
              >
                Wordpress
              </span>
              <span
                class="softStackItem"
              >
                CSS
              </span>
              <span
                class="softStackItem"
              >
                Sass
              </span>
              <span
                class="softStackItem"
              >
                HTML
              </span>
              <span
                class="softStackItem"
              >
                JavaScript
              </span>
              <span
                class="softStackItem"
              >
                TypeScript
              </span>
              <span
                class="softStackItem"
              >
                Webpack
              </span>
              <span
                class="softStackItem"
              >
                Vite
              </span>
              <span
                class="softStackItem"
              >
                React
              </span>
            </div>
          </div>
        </div>
        <div
          class="blockInfo rotate"
        >
          <a
            class="linkImage"
            href="https://rollingscopes.com/"
          >
            <div
              class="imageCommand imageRss"
            />
          </a>
          <div
            class="textInfo"
          >
            <a
              class="titleBlock"
              href="https://rollingscopes.com/"
            >
              <h2>
                Rolling Scopes School
                 
              </h2>
            </a>
            <div>
              RSSchool is an online educational community specializing in web development and programming. It offers comprehensive, well-structured courses designed for both beginners and experienced developers. Known for its high-quality education, RSSchool provides an extensive curriculum that covers everything from basic HTML, CSS, and JavaScript to advanced technologies like React and Node.js. The school is emphasis on project-based learning allows students to build real-world projects and create impressive portfolios.
            </div>
            <div>
              RSSchool thrives on a vibrant community of volunteers, including seasoned developers and alumni who mentor and support new learners.
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
